src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -5195,6 +5271,14 @@ pub fn resolve_type_name(type_name: &TypeName, not_null: bool) -> Result<TableDa
     Ok(data_type)
 }
 
+pub fn resolve_type_name_udf(type_name: &TypeName) -> Result<TableDataType> {
```

Comment:
The logic for the compound type resolve_type_name_udf can be a bit unusual, is this intentional or accidental?
For example, Array(String) equals Array(String NOT NULL) NULL instead of Array(String NULL) NULL.

Reply:
```
root@localhost:8000/default> create table xx (a array(string),b array(string null),c array(string not null));

CREATE TABLE xx (
    a ARRAY(STRING), b ARRAY(STRING NULL), c ARRAY(STRING NOT NULL)
)

0 row written in 0.056 sec. Processed 0 row, 0 B (0 row/s, 0 B/s)

root@localhost:8000/default> show create table xx;

SHOW CREATE TABLE xx

-[ RECORD 1 ]-----------------------------------
       Table: xx
Create Table: CREATE TABLE xx (
  a ARRAY(STRING) NULL,
  b ARRAY(STRING) NULL,
  c ARRAY(STRING) NULL
) ENGINE=FUSE

1 row showcreate in 0.037 sec. Processed 0 row, 0 B (0 row/s, 0 B/s)

root@localhost:8000/default> select * from system.columns where table = 'xx';

SELECT * FROM system.columns WHERE (table = 'xx')

┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  name  │ database │  table │                type               │   data_type   │ default_kind │ default_expression │ is_nullable │ comment │
│ String │  String  │ String │               String              │     String    │    String    │       String       │    String   │  String │
├────────┼──────────┼────────┼───────────────────────────────────┼───────────────┼──────────────┼────────────────────┼─────────────┼─────────┤
│ a      │ default  │ xx     │ Nullable(Array(Nullable(String))) │ ARRAY(STRING) │              │                    │ YES         │         │
│ b      │ default  │ xx     │ Nullable(Array(Nullable(String))) │ ARRAY(STRING) │              │                    │ YES         │         │
│ c      │ default  │ xx     │ Nullable(Array(String))           │ ARRAY(STRING) │              │                    │ YES         │         │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
3 rows read in 0.130 sec. Processed 3 rows, 452 B (23.08 rows/s, 3.39 KiB/s)

```
There is a problem with the SHOW CREATE TABLE.

Reply:
Let's fix it.